src/components/fields/Dropdown.js

```diff
@@ -6,16 +6,31 @@ import {connectToContainer} from 'lib';
 
 export class UnconnectedDropdown extends Component {
   render() {
+    const {multiValued, fullValue, attr} = this.props;
     let placeholder;
-    if (this.props.multiValued) {
-      placeholder = this.props.fullValue;
+    if (multiValued) {
+      placeholder = fullValue;
+    }
+
+    let options = this.props.options;
+
+    if (
+      attr === 'geo.projection.type' &&
+      this.context.fullLayout &&
+      this.context.fullLayout.geo
+    ) {
+      if (this.context.fullLayout.geo.scope === 'usa') {
+        options = options.usa;
+      } else {
+        options = options.other;
+      }
     }
 
     return (
       <Field {...this.props}>
         <DropdownWidget
           backgroundDark={this.props.backgroundDark}
-          options={this.props.options}
+          options={options}
           value={this.props.fullValue}
```

Comment:
if you're pulling stuff out of `props` you may as well reuse them here :)

Reply:
oh forgot about this one : ) thanks!

Reply:
np. For what it's worth, I'm not a huge fan of giant `const {...} = this.props;` everywhere... I'm much happier just having `this.props.x` everywhere as it's less mental indirection when reading the code. Basically if I see `this.props.x` I *know* it's from props and hasn't been locally mutated. If I see just `x` I have to look up higher in the code to know where it came from... Is it `const`? Was it `let`ed and mutated 9 times? etc... :)